src/api/artifact/abstractor/resolver/chart/chart.go

```diff
@@ -86,7 +90,61 @@ func (r *resolver) ResolveMetadata(ctx context.Context, manifest []byte, artifac
 }
 
 func (r *resolver) ResolveAddition(ctx context.Context, artifact *artifact.Artifact, addition string) (*resolv.Addition, error) {
-	// TODO implement
+	if addition != AdditionTypeValues && addition != AdditionTypeReadme && addition != AdditionTypeDependencies {
+		return nil, ierror.New(nil).WithCode(ierror.BadRequestCode).
+			WithMessage("addition %s isn't supported for %s", addition, ArtifactTypeChart)
+	}
+
+	repository, err := r.repoMgr.Get(ctx, artifact.RepositoryID)
+	if err != nil {
+		return nil, err
+	}
+	_, content, err := r.blobFetcher.FetchManifest(repository.Name, artifact.Digest)
+	if err != nil {
+		return nil, err
+	}
+	manifest := &v1.Manifest{}
+	if err := json.Unmarshal(content, manifest); err != nil {
+		return nil, err
+	}
+
+	for _, layer := range manifest.Layers {
+		// chart do have two layers, one is config, we should resolve the other one.
+		layerDgst := layer.Digest.String()
+		if layerDgst != manifest.Config.Digest.String() {
+			content, err = r.blobFetcher.FetchLayer(repository.Name, layerDgst)
+			if err != nil {
+				return nil, err
+			}
+			chartDetails, err := r.chartOperator.GetDetails(content)
+			if err != nil {
+				return nil, err
+			}
+
+			var additionContent []byte
+			var additionContentType string
+
+			switch addition {
+			case AdditionTypeValues:
+				additionContent = []byte(chartDetails.Files[AdditionTypeValues])
+				additionContentType = "text/plain; charset=utf-8"
```

Comment:
Use `text/yaml`? There is no standard for yaml yet

Reply:
let's keep it